### PR TITLE
Tcl overview

### PR DESCRIPTION
**Pull request by pooryorick**
Here is a quick guide to Tcl.

### CODE REVIEW

File: tcl.html.markdown [hunk 1]
```diff
@@ -0,0 +1,377 @@
...
+portable functionality for loading dynamic libraries, string formatting and
+code conversion, filesystem operations, network operations, and more.
+
+Tcl is a pleasure to program in.  Its discipline of exposing all programmatic
+functionality as commands, including things like loops and mathematical
+operations that are usually baked into the syntax of other languages, allows it
+to fade into the background of whatever domain-specific functionality a project
+needs.  Its design of exposing all values as strings, while internally caching
+a structured representation, bridges the world of scripting and systems
+programming in the best way.  Even Lisp is more syntactically heavy than Tcl.
```
Comment (vendethiel) [inline tcl.html.markdown]: values are either strings or lists, if I'm not mistaken

File: tcl.html.markdown [hunk 2]
```diff
@@ -0,0 +1,377 @@
...
+
+
+#semicolon also delimits commands
+
+set greeting1 Sal; set greeting2 ut; set greeting3 ations 
+
+
+# Dollar sign introduces variable substitution
+
+set greeting $greeting1$greeting2
```
Comment (vendethiel) [inline tcl.html.markdown]: maybe add a note saying you don't need quotes around it?

File: tcl.html.markdown [hunk 3]
```diff
@@ -0,0 +1,377 @@
...
+
+
+# Bracket introduces command substitution
+
+set greeting $greeting[set greeting3]
```
Comment (vendethiel) [inline tcl.html.markdown]: if you havn't done any Tcl before, this is confusing. Maybe add a note explaing that `set` with only one argument returns the value?

File: tcl.html.markdown (hunk 1 above)
Comment (pooryorick) [inline tcl.html.markdown]: All values are strings.  Any string that conforms to list formatting requirements *is* a list. See http://wiki.tcl.tk/3018

File: tcl.html.markdown (hunk 3 above)
Comment (pooryorick) [inline tcl.html.markdown]: Done.

File: tcl.html.markdown [hunk 4]
```diff
@@ -0,0 +1,446 @@
+---
+language: Tcl
+contributors:
```
Comment (levibostian) [inline tcl.html.markdown]: Above this line you should have ```filename: learntcl.tcl``` (I'm not sure what the file extension for a TCL file is, I am just guessing .tcl so change that if not correct.)

Comment (pooryorick) [inline tcl.html.markdown]: Done.

**Comment by vendethiel** [comment]
Thanks!

**Pull request merged**